keyboards/edinburgh41/keymaps/default/keymap.c

```diff
@@ -0,0 +1,169 @@
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_edinburgh41(
+    KC_TAB,   KC_Q,     KC_W,     KC_E,     KC_R,      KC_T,               KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_BSPC,
+    KC_LCTL,  KC_A,     KC_S,     KC_D,     KC_F,      KC_G,               KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,  KC_QUOT,
+    KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,      KC_B,               KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,  RSFT_T(KC_ENT),
+                                            KC_LALT,   LOWER,    KC_SPC,   RAISE,    KC_RGUI
+  ),
+
+  [_LOWER] = LAYOUT_edinburgh41(
+    _______,  KC_EXLM,  KC_AT,    KC_HASH,  KC_DLR,    KC_PERC,            KC_CIRC,  KC_AMPR,  KC_ASTR,  KC_LPRN,  KC_RPRN,  KC_DEL,
+    _______,  KC_UNDS,  KC_PLUS,  KC_LCBR,  KC_RCBR,   KC_PIPE,            KC_LEFT,  KC_DOWN,  KC_UP,    KC_RGHT,  KC_GRV,   KC_TILD,
+    _______,  KC_ESC,   KC_LGUI,  KC_LALT,  KC_CAPS,   KC_DQUO,            KC_HOME,  KC_END,   KC_PGUP,  KC_PGDN,  KC_PSCR,  RSFT_T(KC_SPC),
+                                            _______,   _______,  KC_ENT,   _______,  _______
+  ),
+
+  [_RAISE] = LAYOUT_edinburgh41(
+    _______,  KC_1,     KC_2,     KC_3,     KC_4,      KC_5,               KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_DEL,
+    _______,  KC_MINS,  KC_EQL,   KC_LBRC,  KC_RBRC,   KC_BSLS,            KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,
+    _______,  KC_ESC,   KC_RGUI,  KC_RALT,  KC_CAPS,   KC_QUOT,            KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,
+                                            _______,   _______,  KC_BSPC,  _______,  _______
+  ),
+
+  [_ADJUST] = LAYOUT_edinburgh41(
+    RGB_VAI,   RGB_SAI, RGB_HUI,  RGB_MOD,  XXXXXXX,   RGB_TOG,            XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+    RGB_VAD,   RGB_SAD, RGB_HUD,  RGB_RMOD, XXXXXXX,   XXXXXXX,            XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+    XXXXXXX,   XXXXXXX, XXXXXXX,  XXXXXXX,  XXXXXXX,   XXXXXXX,            QK_BOOT,    XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+                                            _______,   _______,  XXXXXXX,  _______,  _______
+  ),
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+// Thumbstick keymap, change KC_XXX to whatever you need
+
+#define THUMBSTICK_RIGHT_TAP KC_RIGHT
+#define THUMBSTICK_LEFT_TAP KC_LEFT
+#define THUMBSTICK_UP_TAP KC_UP
+#define THUMBSTICK_DOWN_TAP KC_DOWN
+
+// Thumbstick code, no customisation needed
+
+bool cursor_mode = false;
+bool scrolling_mode = false;
+bool tapping_mode = false;
+
+// tracks if thumbstick was released
+bool returned_to_zero = true;
+
+// tracks how many times mouse_report.x/y have been read zero in succession
+uint16_t zero_reads = 0;
+
+// set mode depending on layer
+layer_state_t layer_state_set_kb(layer_state_t state) {
```

Comment:
Cant have `_kb` funcs at keymap level
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```